sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Listeners/HttpRequestProcessor.cs

```diff
@@ -0,0 +1,110 @@
+        private BlobTriggerQueueWriter _blobTriggerQueueWriter;
+
+        public HttpRequestProcessor(
+            SharedQueueWatcher sharedQueueWatcher,
+            IHostIdProvider hostIdProvider,
+            QueueServiceClientProvider queueServiceClientProvider,
+            ILoggerFactory loggerFactory)
+        {
+            _sharedQueueWatcher = sharedQueueWatcher;
+            _hostIdProvider = hostIdProvider;
+            _queueServiceClientProvider = queueServiceClientProvider;
+            _logger = loggerFactory.CreateLogger<HttpRequestProcessor>();
+        }
+
+        public void RegisterHttpEnpoint(ExtensionConfigContext context)
+        {
+            //Debugger.Break();
+#pragma warning disable CS0618 // Type or member is obsolete
+            Uri url = context.GetWebhookHandler();
+#pragma warning restore CS0618 // Type or member is obsolete
+            _logger.LogInformation($"registered http endpoint = {url?.GetLeftPart(UriPartial.Path)}");
+        }
+
+        public async Task<HttpResponseMessage> ProcessHttpRequestAsync(HttpRequestMessage req, CancellationToken cancellationToken)
+        {
+            //Debugger.Break();
+            var functionName = HttpUtility.ParseQueryString(req.RequestUri.Query)["functionName"];
+
+            IEnumerable<string> eventTypeHeaders = null;
+            string eventTypeHeader = null;
+            if (req.Headers.TryGetValues("aeg-event-type", out eventTypeHeaders))
+            {
+                eventTypeHeader = eventTypeHeaders.First();
+            }
+
+            return await EventGrid.HttpRequestProcessor.ProcessAsync(req, functionName, _logger, ProcessEventsAsync, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        private async Task<HttpResponseMessage> ProcessEventsAsync(JArray events, string functionName, CancellationToken cancellationToken)
+        {
+            if (_blobTriggerQueueWriter == null)
+            {
+                await InitializeWriterAsync(cancellationToken).ConfigureAwait(false);
+            }
+
+            foreach (JObject jo in events)
+            {
+                BlobTriggerMessage blobTriggerMessage = GetBlobTriggerMessage(jo, functionName);
+                await _blobTriggerQueueWriter.EnqueueAsync(blobTriggerMessage, cancellationToken).ConfigureAwait(false);
+            }
+
+            return new HttpResponseMessage(HttpStatusCode.Accepted);
+        }
+
+
+        private async Task InitializeWriterAsync(CancellationToken cancellationToken)
+        {
+            string hostId = await _hostIdProvider.GetHostIdAsync(cancellationToken).ConfigureAwait(false);
+            string hostBlobTriggerQueueName = HostQueueNames.GetHostBlobTriggerQueueName(hostId);
+            var hostBlobTriggerQueue = _queueServiceClientProvider.GetHost().GetQueueClient(hostBlobTriggerQueueName);
+
+            _blobTriggerQueueWriter = new BlobTriggerQueueWriter(hostBlobTriggerQueue, _sharedQueueWatcher);
+        }
```

Comment:
Can we DI this? I.e. inject ready to use blobTriggerQueueWritter to this class? We might need a factory like https://github.com/Azure/azure-sdk-for-net/blob/785ae16dc06fc15e4de4e326e118a88ad9dee96c/sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Listeners/BlobListenerFactory.cs#L149 .

Resolving blob queue name is independent from the strategy we listen to blob changes, so there's an opportunity to reduce code duplication as well.

Reply:
`BlobTriggerQueueWriterFactory` was added.